src/components/validation/validators/component_spec/sources.rs

```diff
@@ -207,28 +208,13 @@ fn validate_component_received_bytes_total(
 ) -> Result<Vec<String>, Vec<String>> {
     let mut errs: Vec<String> = Vec::new();
 
-    let metrics = filter_events_by_metric_and_component(
+    let metrics = filter_events_by_metric_and_component_with_errors(
         telemetry_events,
-        SourceMetrics::ReceivedBytesTotal,
+        &SourceMetrics::ReceivedBytesTotal,
         TEST_SOURCE_NAME,
     )?;
 
-    let mut metric_bytes: f64 = 0.0;
-    for m in metrics {
-        match m.value() {
-            vector_core::event::MetricValue::Counter { value } => {
-                if let MetricKind::Absolute = m.data().kind {
-                    metric_bytes = *value
-                } else {
-                    metric_bytes += value
-                }
-            }
-            _ => errs.push(format!(
-                "{}: metric value is not a counter",
-                SourceMetrics::ReceivedBytesTotal,
-            )),
-        }
-    }
+    let metric_bytes: f64 = sum_counters(SourceMetrics::ReceivedBytesTotal, &metrics)?;
```

Comment:
nit: the explicit type declaration isn't necessary for the cases where you are using the returned `f64` type without casting.

Reply:
fixed in https://github.com/vectordotdev/vector/pull/16854/commits/399af00b7d2e5c77b837bace9c02402c3954c676